src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -883,22 +885,24 @@ object Parser {
         declaredType
 
     lazy val predeclaredType: Parser[PPredeclaredType] =
-      "bool" ^^^ PBoolType() |
+      exactWord("bool") ^^^ PBoolType() |
         // signed integer types
-        "rune" ^^^ PRune() |
-        "int8" ^^^ PInt8Type() |
-        "int16" ^^^ PInt16Type() |
-        "int32" ^^^ PInt32Type() |
-        "int64" ^^^ PInt64Type() |
-        "int" ^^^ PIntType() | // 'int' must come after all 'intX' parsers, otherwise parsing will stop at 'int'
+        exactWord("rune") ^^^ PRune() |
+        exactWord("int") ^^^ PIntType() |
+        exactWord("int8") ^^^ PInt8Type() |
+        exactWord("int16") ^^^ PInt16Type() |
+        exactWord("int32") ^^^ PInt32Type() |
+        exactWord("int64") ^^^ PInt64Type() |
         // unsigned integer types
-        "byte" ^^^ PByte() |
-        "uint8" ^^^ PUInt8Type() |
-        "uint16" ^^^ PUInt16Type() |
-        "uint32" ^^^ PUInt32Type() |
-        "uint64" ^^^ PUInt64Type() |
-        "uintptr" ^^^ PUIntPtr() |
-        "uint" ^^^ PUIntType()
+        exactWord("byte") ^^^ PByte() |
+        exactWord("uint") ^^^ PUIntType() |
+        exactWord("uint8") ^^^ PUInt8Type() |
+        exactWord("uint16") ^^^ PUInt16Type() |
+        exactWord("uint32") ^^^ PUInt32Type() |
+        exactWord("uint64") ^^^ PUInt64Type() |
+        exactWord("uintptr") ^^^ PUIntPtr()
+
+    private def exactWord(s: String): Regex = (s ++ "\\b").r
```

Comment:
Why do you only use `\b` at the end of `s`? Doesn‘t it work if you add it on both sides?

Reply:
Yes, it does work if I add it on both sides.

Reply:
I was trying to minimize the regex to be matched on but i do not think that it makes a difference in this particular case. I will change that

Reply:
It might be true that in our parser "blaint" will never be parsed as "bla" "int" but nevertheless (and given the function name) I prefer checking for word boundaries on both sides